### ISSUE TITLE
Storybook for react makes webpack crash

### ISSUE THREAD

**Issue by german-bortoli** [storybookjs/storybook#10304]
**Versions:**
 - Node: v10.16.3
 - NPM: v6.9.0

**Libraries Versions:**
 - react-scripts: 3.4.1
 - @storybook/react: 5.3.18
 - @storybook/preset-create-react-app: 2.1.1

**Result**

Having different webpack versions my react application won't start

```
├─┬ @storybook/react@5.3.18
│ ├─┬ @storybook/core@5.3.18
│ │ ├─┬ corejs-upgrade-webpack-plugin@2.2.0
│ │ │ └── webpack@4.42.1  deduped
│ │ └── webpack@4.42.1  deduped
│ └── webpack@4.42.1
└─┬ react-scripts@3.4.1
  └── webpack@4.42.0
```

**Comment by TripleSpeeder** [comment]
Also happens for me with newer node/npm versions:
* Node: v10.19.0
* NPM: v6.14.4

Yarn managed resolving the webpack version better (it used 4.42.0 as required by react-scripts) and my app can start. However it failed to create a production build, and i did not yet bother checking in detail what went wrong (pretty sure it is an unrelated issue though).
@german-bortoli so you might have more luck using yarn until this is solved :-)

**Comment by german-bortoli** [comment]
@TripleSpeeder I tried using Yarn also, and did not went very good, some problems got fixed but webpack was not able to transpile Typescript pretty well, for example, the props from redux couldn´t be mapped properly.

**Comment by shilman** [comment]
@german-bortoli can you share a reproduction repo?

**Comment by TripleSpeeder** [comment]
@shilman You can also check the development branch of https://github.com/TripleSpeeder/token-allowance-checker. As a workaround i added a fixed dependency to webpack 4.42.0 in package.json. If you remove this dependency you will see the problem.

Overall I think this is more an issue with npm and the way it resolves dependencies. Don't think storybook code is actual responsible, it's just a coincidence it's appearing here.

**Comment by german-bortoli** [comment]
@shilman no, I can't share the code repo :s sorry
